lib/ansible/plugins/loader.py

```diff
@@ -279,7 +282,91 @@ def add_directory(self, directory, with_subdir=False):
                 self._paths = None
                 display.debug('Added %s to loader search path' % (directory))
 
-    def _find_plugin(self, name, mod_type='', ignore_deprecated=False, check_aliases=False):
+    def _find_fq_plugin(self, fq_name, extension):
+        # prefix our extension Python namespace if it isn't already there
+        if not fq_name.startswith('ansible_collections.'):
+            fq_name = 'ansible_collections.' + fq_name
+
+        splitname = fq_name.rsplit('.', 1)
+        if len(splitname) != 2:
+            raise ValueError('{0} is not a valid namespace-qualified plugin name'.format(to_native(fq_name)))
+
+        package = splitname[0]
+        resource = splitname[1]
+
+        append_plugin_type = self.class_name or self.subdir
+
+        if append_plugin_type:
+            # only current non-class special case, module_utils don't use this loader method
+            if append_plugin_type == 'library':
+                append_plugin_type = 'modules'
+            else:
+                append_plugin_type = get_plugin_class(append_plugin_type)
+            package += '.plugins.{0}'.format(append_plugin_type)
+
+        if extension:
+            resource += extension
+
+        #plugin_content = None
+
+        # try:
+        #     # FIXME: now that we're doing custom loader magic, get rid of the get_data reverse-engineering and directly
+        #     # consult the packages and/or custom loaders to just get the file path instead of loading here
+        #     # longer-term, maybe optimize everything to use get_data instead?
+        #     plugin_content = pkgutil.get_data(package, resource)
+        #
+        #     if plugin_content is None:
+        #         return None
+        #
+        #     # HACK: if this is a typed loader, ensure there's a class in there
+        #     # NB: this hack should not be necessary for type-specific plugins
+        #     if self.class_name:
+        #         if not b"class %s" % to_bytes(self.class_name) in plugin_content:
+        #             return None
+        # except IOError:
+        #     if extension:
+        #         return None  # extension was specified and we didn't find it, move on
+
+        pkg = sys.modules.get(package)
+        if not pkg:
+            # FIXME: there must be cheaper/safer way to do this
+            pkg = importlib.import_module(package)
```

Comment:
Note: not having `action` directory in the collection raises an exception.
```
- hosts: localhost
  gather_facts: no
  tasks:
    - mynamespace.mycollection.ping:
```

```
mkrizek@t470s:~/src/ansible [pr-52194] $ tree ~/.ansible/content/ansible_collections/
/home/mkrizek/.ansible/content/ansible_collections/
├── __init__.py
└── mynamespace
    ├── __init__.py
    └── mycollection
        ├── __init__.py
        └── plugins
            ├── __init__.py
            └── modules
                ├── __init__.py
                └── ping.py

4 directories, 6 files
```

```
mkrizek@t470s:~/src/ansible [pr-52194] $ ansible-playbook issues/playbook_collections.yml -vvv
ansible-playbook 2.8.0.dev0
  config file = /home/mkrizek/src/ansible/ansible.cfg
  configured module search path = [u'/home/mkrizek/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']                                                                                              
  ansible python module location = /home/mkrizek/src/ansible/lib/ansible
  executable location = /home/mkrizek/src/ansible/bin/ansible-playbook
  python version = 2.7.15 (default, Oct 15 2018, 15:26:09) [GCC 8.2.1 20180801 (Red Hat 8.2.1-2)]
Using /home/mkrizek/src/ansible/ansible.cfg as config file
host_list declined parsing /home/mkrizek/src/ansible/hosts as it did not pass it's verify_file() method
script declined parsing /home/mkrizek/src/ansible/hosts as it did not pass it's verify_file() method
auto declined parsing /home/mkrizek/src/ansible/hosts as it did not pass it's verify_file() method
Parsed /home/mkrizek/src/ansible/hosts inventory source with ini plugin

PLAYBOOK: playbook_collections.yml **********************************************************************
1 plays in issues/playbook_collections.yml

PLAY [localhost] ****************************************************************************************
META: ran handlers
ERROR! Unexpected Exception, this is probably a bug: No module named action
the full traceback was:

Traceback (most recent call last):
  File "/home/mkrizek/src/ansible/bin/ansible-playbook", line 110, in <module>
    exit_code = cli.run()
  File "/home/mkrizek/src/ansible/lib/ansible/cli/playbook.py", line 123, in run
    results = pbex.run()
  File "/home/mkrizek/src/ansible/lib/ansible/executor/playbook_executor.py", line 170, in run
    result = self._tqm.run(play=play)
  File "/home/mkrizek/src/ansible/lib/ansible/executor/task_queue_manager.py", line 248, in run
    play_return = strategy.run(iterator, play_context)
  File "/home/mkrizek/src/ansible/lib/ansible/plugins/strategy/linear.py", line 245, in run
    action = action_loader.get(task.action, class_only=True)
  File "/home/mkrizek/src/ansible/lib/ansible/plugins/loader.py", line 545, in get
    path = self.find_plugin(name, collection_list=collection_list)
  File "/home/mkrizek/src/ansible/lib/ansible/plugins/loader.py", line 496, in find_plugin
    plugin = self._find_plugin(name, mod_type=mod_type, ignore_deprecated=ignore_deprecated, check_aliases=check_aliases, collection_list=collection_list)
  File "/home/mkrizek/src/ansible/lib/ansible/plugins/loader.py", line 408, in _find_plugin
    raise Exception('; '.join(errors))
Exception: No module named action
```

Reply:
I think this may be the same issue that https://github.com/alikins/ansible/commit/4e5168b7d69cf907fe0fa21ffeb932cb82f290d6 from https://github.com/alikins/ansible/commits/pr-52194-fixes addresses.